test/options.cpp

```diff
@@ -473,3 +473,30 @@ TEST_CASE("std::optional", "[optional]") {
   CHECK(*optional == "foo");
 }
 #endif
+
+TEST_CASE("Unrecognised options", "[options]") {
+  cxxopts::Options options("unknown_options", " - test basic options");
+
+  options.add_options()
+    ("long", "a long option")
+    ("s,short", "a short option");
+
+  Argv av({
+    "unknown_options",
+    "--long",
+    "-su",
+    "--unknown",
+  }); 
+
+  const char** argv = av.argv();
+  auto argc = av.argc();
+
+  SECTION("Default behaviour") {
+    CHECK_THROWS_AS(options.parse(argc, argv), cxxopts::option_not_exists_exception);
+  }
+
+  SECTION("After allowing unrecognised options") {
+    options.allow_unrecognised_options();
+    CHECK_NOTHROW(options.parse(argc, argv));
```

Comment:
It might be good to check that `argc` is still 4.

Reply:
Well, right now ``argc`` won't be 4 but will be changed just as before by changes (so I believe it changes to 1 in the test case). I wasn't sure about the semantics of modifying ``argc``'s value so I didn't touch that behavior. It doesn't seem to be documented anywhere.

Of course I can change it back to the original value in case the parsing detects any unspecified options. Is that the behavior you would recommend?

Reply:
It makes sense for it to be 4 if it's skipping unrecognised options. Although I just realised that I'm not sure if the code that consumes arguments will work with this.